instrumentation/src/main/java/io/opentelemetry/android/instrumentation/startup/SdkInitializationEvents.java

```diff
@@ -43,4 +44,9 @@ public void slowRenderingDetectorInitialized() {
     public void crashReportingInitialized() {
         // TODO: Build me "crashReportingInitialized"
     }
+
+    @Override
+    public void spanExporterInitialized(SpanExporter spanExporter) {
```

Comment:
Not related to this PR though when I was changing this class I thought that maybe we don't need this implementation of `InitializationEvents` and instead we can just add a public setter [in this class](https://github.com/open-telemetry/opentelemetry-android/blob/6de29672a2bf44cc36b651bdb12b7f5228856333/instrumentation/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java#L85) so that, if a user needs to enable initialization events, they can just set their own implementation in there? That way we wouldn't need to define [this config flag](https://github.com/open-telemetry/opentelemetry-android/blob/6de29672a2bf44cc36b651bdb12b7f5228856333/instrumentation/src/main/java/io/opentelemetry/android/config/OtelRumConfig.java#L27) either, as the default impl would be the noop one.

Reply:
That sounds like a good idea

Reply:
> It probably doesn't help that the existing implementation is just a stub. I think I would _not_ want 20 different users to have 20 different ways of formulating these various events. In other words, ideally one day in the future we will have semconv for these events, and I'd hate for `opentelemetry-android` users to have to go read the semconv in order to build an implementation themselves. We should provide that implementation for them, even if we do decide to default these to "off" (no-op). Users that want them shouldn't have to build it.

Definitely having a default impl should be the way to go, although it's quite pointless having a stub for now that also cannot be overridden (since it can also be a great tool for unit tests). I would prefer to provide a default implementation returned by a `InitializationEvents.getDefault()` static function and a setter in `OpenTelemetryRumBuilder` that users (and ourselves from unit tests) could call to enable this feature, since it would be set to the `noop` impl by default, which essentially disables it. Rather than having to create a config flag for it, generally speaking I don't like flags, they can be very useful for small things but are not good for scalability. I also haven't seen them used in OTel Java for example, is mostly interfaces, setters for their implementations, and implementations already created in the same repo that are useful for most use cases and ready to be used so that users can take advantage of them easily.